system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -38,30 +40,51 @@ PeriodicMeasurementNode::PeriodicMeasurementNode(
   publish_period_(publish_period),
   clear_measurements_on_publish_(clear_measurements_on_publish)
 {
-  if (publish_period_ < std::chrono::milliseconds(0)) {
+  if (publish_period_ <= std::chrono::milliseconds(0)) {
     throw std::invalid_argument("publish period cannot be negative");
   }
 }
 
 bool PeriodicMeasurementNode::setupStart()
 {
   assert(measurement_timer_ == nullptr);
+  assert(publish_timer_ == nullptr);
 
   RCLCPP_DEBUG(this->get_logger(), "setupStart: creating measurement_timer_");
   measurement_timer_ = this->create_wall_timer(
-    measurement_period_,
-    std::bind(&PeriodicMeasurementNode::performPeriodicMeasurement, this));
+    measurement_period_, [this]() {this->performPeriodicMeasurement();});
+
+  if (publisher_ == nullptr) {
```

Comment:
Is the publisher check necessary?

Reply:
Also for the restart (`start()` -> `stop()` -> `start()`) case, I wanted to avoid object churn by reusing a previously created publisher.